packages/kernel/src/worker.ts

```
@@ -177,6 +174,11 @@ async function loadPyodideAndPackages() {
     throw new Error(`Neither snapshot nor wheel files are provided.`);
   }
 
+  while (!tokenIsSet) {
+    // Wait for Fusion to send token
+    await new Promise((r) => setTimeout(r, 50));
```

Could we add a max retry check here?

---

Yeah, this leads to silent failure. Will create a task